bundles/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/discovery/BigAssFanDiscoveryService.java

```diff
@@ -116,20 +116,27 @@ protected void stopBackgroundDiscovery() {
     }
 
     private void startListenerJob() {
-        if (listenerJob == null) {
-            terminate = false;
-            logger.debug("Starting discovery listener job in {} seconds", BACKGROUND_DISCOVERY_DELAY);
-            listenerJob = scheduledExecutorService.schedule(listenerRunnable, BACKGROUND_DISCOVERY_DELAY,
-                    TimeUnit.SECONDS);
+        logger.debug("Starting discovery listener job in {} seconds", BACKGROUND_DISCOVERY_DELAY);
+
+        synchronized (this) {
```

Comment:
It would be better to synchronize on some private object instead of `this` because you never know what other logic is also synchronizing on the object instance.

If you do ever want to ever synchronize a whole method on `this` it's easier to add the `synchronized` modifier between `private void` so you do not end up with all the unncessary indentation and 2 more lines of code. :wink:

Reply:
It was a code suggestion from fwolter or cpmeister a year ago i used for PlugwiseHA binding. So i might have to alter that too. i'll have a look at how other bindings solve this. Thank you